### PR TITLE
Add report module to sgx-types

### PR DESCRIPTION
**Pull request by lkatalin**
Here is the basic memory layout for Report. I can add functionality as needed.

### CODE REVIEW

File: sgx-types/src/report.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+//! The REPORT structure is the output of the EREPORT instruction, and must be 512-Byte aligned.
+
+use super::{attr::Attributes, isv, misc::MiscSelect};
+use testing::testaso;
+
+/// This struct is separated out from the Report to be usable by the Quote struct.
+/// Table 38-21
+#[derive(Default)]
+#[repr(C)]
+pub struct ReportBody {
```
Comment (npmccallum) [inline sgx-types/src/report.rs]:
```suggestion
pub struct Body {
```

File: sgx-types/src/report.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    /// Data provided by the user and protected by the Report's MAC (Section 38.15.1)
+    pub reportdata: [u16; 32],
+}
+
+/// Table 38-21
+#[derive(Default)]
+#[repr(C, align(512))]
+pub struct Report {
+    /// The body of the Report
+    reportbody: ReportBody,
```
Comment (npmccallum) [inline sgx-types/src/report.rs]:
```suggestion
    pub reportbody: ReportBody,
```

Is this supposed to be private?

File: sgx-types/src/report.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    pub mrsigner: [u8; 32],
+    reserved2: [u32; 24],
+    /// Product ID of the enclave
+    pub isvprodid: isv::ProdId,
+    /// Security version number of the enclave
+    pub isvsvn: isv::Svn,
+    reserved3: [u32; 15],
+    /// Data provided by the user and protected by the Report's MAC (Section 38.15.1)
+    pub reportdata: [u16; 32],
+}
```
Comment (npmccallum) [inline sgx-types/src/report.rs]: I suspect we probably need padding at the end. This struct will likely be signed. So we need to make sure that any trailing bytes are properly zeroed.

File: sgx-types/src/report.rs (hunk 1 above)
Comment (npmccallum) [reply sgx-types/src/report.rs]: We don't prefix types by the module name.

File: sgx-types/src/report.rs (hunk 3 above)
Comment (lkatalin) [reply sgx-types/src/report.rs]: Very good point; I just thought of this during the standup, based on yesterday's PR.

File: sgx-types/src/report.rs (hunk 2 above)
Comment (lkatalin) [reply sgx-types/src/report.rs]: I don't think so. That was a mistake.

File: sgx-types/src/report.rs (hunk 3 above)
Comment (npmccallum) [reply sgx-types/src/report.rs]: I misread this code. It looks like `reportdata` ends at the end of the struct. So there's no padding needed here.

**Comment by npmccallum** [comment]
@lkatalin Also, I personally try to avoid speaking about filesystem paths in commit messages. This can be misleading when files move around later. But this is just a nitpick.

**Pull request merged**